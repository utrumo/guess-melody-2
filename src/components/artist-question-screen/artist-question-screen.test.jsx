import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen.jsx';
import createAudioNodeMock from '../../mocks/audio-node-mock';

it(`ArtistQuestionScreen correctly renders after relaunch`, () => {
  const onUserAnswer = jest.fn();
  const mockQuestion = {
    type: `artist`,
    song: {
      artist: `Quincas Moreira`,
      src: `https://es31-server.appspot.com/guess-melody/static/music/Blue_Whale.mp3`
    },
    answers: [
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
        artist: `Quincas Moreira`
      },
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
        artist: `Density & Time`
      },
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
        artist: `Endless Love`
      }
    ]
  };
  const questionIndex = 2;

  const artistQuestionScreen = renderer
    .create(
        <ArtistQuestionScreen
          onAnswer={onUserAnswer}
          question={mockQuestion}
          screenIndex={questionIndex}
        />,
        createAudioNodeMock
    )
    .toJSON();

  expect(artistQuestionScreen).toMatchSnapshot();
});
