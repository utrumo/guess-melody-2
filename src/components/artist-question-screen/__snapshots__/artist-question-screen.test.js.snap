// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly renders after relaunch 1`] = `
<section
  className="game game--artist"
>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://es31-server.appspot.com/guess-melody/static/music/Blue_Whale.mp3"
          />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          checked={false}
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="answer-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="Quincas Moreira"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
          />
          Quincas Moreira
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          checked={false}
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="answer-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Density & Time"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
          />
          Density & Time
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          checked={false}
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="answer-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Endless Love"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
          />
          Endless Love
        </label>
      </div>
    </form>
  </section>
</section>
`;
